lib/network/encoding.go

```diff
@@ -120,8 +125,9 @@ func (am *NetworkMessage) SetError(err error) {
 }
 
 var typeRegistry = make(map[uuid.UUID]reflect.Type)
+var registryLock = new(sync.Mutex)
```

Comment:
Maybe, put this in a struct, which embeds a mutex and only is accessible through functions for reading and writing to the map `typeRegistry` (and which handles the (un)locking)